internal/locald/rootmanager/tp_monitor.go

```diff
@@ -128,6 +129,23 @@ func (mon *tpMonitor) checkTunnelProxyAccess(ctx context.Context) bool {
 			restartSvcs = true
 		}
 	}
+	if !restartSvcs {
+		// the grpc check for connecting to the tunnel proxy does not suffice
+		// because it has built-in retries and may re-use a connection while
+		// we are unable to establish a new connection.  So, we also check
+		// the agent health endpoint
+		cli := &http.Client{
+			Transport: &http.Transport{},
+		}
```

Comment:
Maybe it is worth configuring a timeout here?

Reply:
I added a timeout and made a couple of other changes and fixes:
1. access agent-metrics (we don't have any health endpoints exposed, this was always restarting)
2. make the health more pessimistic, return unhealthy when restart is needed